packages/pagination/src/elements/Pagination.js

```diff
@@ -255,8 +275,44 @@ export default class Pagination extends ControlledComponent {
     return (
       <PaginationContainer
         id={id}
-        focusedKey={focusedKey}
-        selectedKey={currentPage}
+        rtl={isRtl(this.props)}
+        focusedItem={focusedKey}
+        selectedItem={currentPage}
+        onFocus={item => {
+          this.setControlledState({ focusedKey: item });
+        }}
+        onSelect={item => {
+          const { totalPages, onChange } = this.props;
+          let updatedCurrentPage = item;
+          let updatedFocusedKey = focusedKey;
+
+          if (updatedCurrentPage === PREVIOUS_KEY && currentPage > 1) {
+            updatedCurrentPage = currentPage - 1;
+
+            // Must manually change focusedKey once PreviousPage is no longer visible
+            if (updatedCurrentPage === 1 && focusedKey === PREVIOUS_KEY) {
+              updatedFocusedKey = 1;
+            }
+          } else if (updatedCurrentPage === NEXT_KEY && currentPage < totalPages) {
+            updatedCurrentPage = currentPage + 1;
+
+            // Must manually change focusedKey once NextPage is no longer visible
+            if (updatedCurrentPage === totalPages && updatedFocusedKey === NEXT_KEY) {
+              updatedFocusedKey = totalPages;
+            }
+          } else if (typeof updatedCurrentPage === 'number') {
+            // updatedCurrentPage = updatedCurrentPage;
```

Comment:
What's going on here? Seems like something eslint should've caught.

Reply:
Removed.